felix/bpf-gpl/tc.c

```diff
@@ -365,11 +371,13 @@ static CALI_BPF_INLINE int calico_tc(struct __sk_buff *skb)
 		ctx.state->nat_dest.port = 0;
 	}
 
+syn_force_policy:
```

Comment:
Still looks like there are some lines that you jump over that are involved in setting up the state for the policy program.  Don't we need to fill in `nat_dest.*`, `pol_rc` and any flags to match the connection?

In general, there could be pre-DNAT policy and normal policy so I think all the fields would need to be filled in.

Reply:
I think you are right. Things are filled up correctly for the `calico_tc_skb_accepted` but policy uses the `state` and that isn't filled up correctly. I am working on a UT test.

Reply:
I pushed a fix and a test